### ISSUE TITLE
Provider AWS: aws_db_subnet_group keeps reloading on upercase name

### ISSUE THREAD

**Issue by mzupan** [hashicorp/terraform#1750]
Using the following version

```
Terraform v0.5.0-dev (fa85e6b7692b816df0c784f8339e0446bb3a4dc9+CHANGES)
```

This keeps reloading

```
resource "aws_db_subnet_group" "web-db-dmz" {
    name = "Web - DB - DMZ"
    description = "A subnet group for any db instances that need a public IP"
    subnet_ids = ["${aws_subnet.mgmt-dmz-b.id}", "${aws_subnet.mgmt-dmz-c.id}"]
}
```

A plan output is

```
-/+ aws_db_subnet_group.web-db-dmz
    description:           "A subnet group for any db instances that need a public IP" => "A subnet group for any db instances that need a public IP"
    name:                  "web - db - dmz" => "Web - DB - DMZ" (forces new resource)
    subnet_ids.#:          "2" => "2"
    subnet_ids.3799398419: "subnet-4cf5863b" => "subnet-4cf5863b"
    subnet_ids.401784199:  "subnet-b8388193" => "subnet-b8388193"
```

Seems somewhere is making the name lower case

**Comment by catsby** [comment]
Both Terraform and AWS are lower casing things here; AWS stores `DBSubnetGroupName` as a lower case string:

- http://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBSubnetGroup.html

Terraform lower cases the `name` for comparison, and if a match is found, we use the name from AWS instead of the `name` you've provided. 
- https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_db_subnet_group.go#L95-L110

I'm not sure if we should 1) document that `name` should be lower case, or 2) use the `name` provided and hide the lower casing in the background. Leaning toward 2, what do you think?

**Comment by mzupan** [comment]
@catsby I think it terraform should just lower case the name in that case. I could see cases where people over look that in the docs and a lot of issues get created around it.

**Comment by catsby** [comment]
Terraform _is_ lowercasing the nam, in the state file. 
It's not modifying your configuration file, which is where the change is being detected.

**Comment by catsby** [comment]
Proposal fix in #1753

**Comment by catsby** [comment]
Merged #1753, thanks for reporting this!
